main/src/ca/uwaterloo/flix/language/errors/WeederError.scala

```diff
@@ -896,4 +896,61 @@ object WeederError {
          |""".stripMargin
     })
   }
+
+
+  /**
+    * An error raised to indicate that a loop does not iterate over any collection.
+    */
+  case class LoopOverNoCollection(loc: SourceLocation) extends WeederError {
+    def summary: String = "Loop does not contain a collection to loop over."
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> No iterable collection in loop.
+         |
+         |${code(loc, "Loop does not iterate over collection")}
+         |
+         |""".stripMargin
+    }
+
+    def explain(formatter: Formatter): Option[String] = Some({
+      s"""A loop requires at least one collection with an instance of
+         |the Iterable type class on it.
+         |
+         |A minimal loop is written as follows:
+         |
+         |foreach (x <- xs) yield x
+         |
+         |""".stripMargin
+    })
+  }
+
+  /**
+    * An error raised to indicate that a loop has guards before
+    * first iterable collection.
+    */
+  case class LoopGuardsBeforeCollection(loc: SourceLocation) extends WeederError {
```

Comment:
Do we need these as separate? Why not just "loop thingy must start with collection"?

Reply:
I considered the same thing but decided maybe it would be "friendlier" to have more fine grained messages. Thoughts?

Reply:
I prefer just one but maybe @magnus-madsen has a preference.